server/src/locale-messages.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * This is written to work with the locale js files generated by
+ * bin/build-scripts/ftl-to-js.js. Those files place the content strings of ftl
+ * files into window.l10nMessages.
+ *
+ * It waits a _max_ of about five seconds for the js files to be retrieved.
+ */
+exports.getLocaleMessages = function(locales) {
+  return Promise.all(locales.map(locale => {
+    return new Promise((resolve, reject) => {
+      let wait = 5000; // maybe make configurable?
+      let checkForMessages = () => {
+        if (window && window.l10nMessages && window.l10nMessages[locale]) {
+          let messages = {};
+          messages[locale] = window.l10nMessages[locale];
+          resolve(messages);
+          return;
+        }
+        if (wait > 0) {
+          wait -= 100;
+          setTimeout(checkForMessages, 100);
+          return;
+        }
+        reject(locale);
+      };
+      checkForMessages();
+    });
```

Comment:
You can use the callback referenced earlier (`notifyL10nLoaded`) to avoid the polling here. Though it's still good to account for either ordering (i.e., the locales load first, then this file; or this file loads first then the locales).